pseudopotential_calculator/calculations/adsorbate.py

```diff
@@ -0,0 +1,103 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import matplotlib.pyplot as plt
+import numpy as np
+from ase import Atom, Atoms
+from matplotlib import cm
+
+from pseudopotential_calculator.atoms import Vector, get_basis_vectors, repeat_cell
+from pseudopotential_calculator.calculations.slab import get_height_of_layer
+from pseudopotential_calculator.castep import (
+    get_calculator_atom,
+)
+
+if TYPE_CHECKING:
+    from ase.calculators.castep import Castep
+    from matplotlib.figure import Figure
+
+
+def get_top_position(atoms: Atoms) -> tuple[float, float, float]:
+    positions = atoms.positions  # type: ignore bad lib
+    return tuple(max(positions, key=lambda x: x[2]))  # type: ignore bad lib
+
+
+def get_top_layer_z_vector(atoms: Atoms) -> tuple[float, float, float]:
```

Comment:
this is name is misleading - it has nothing to do with the top layer

Reply:
z = get_height_of_layer(atoms, 0)

I want to get the z vector of the top layer which is the zeroth layer of the slab. Maybe get_first_slab_layer_z_vector?